src/dex/balancer-v2/balancer-v2.ts

```diff
@@ -464,7 +486,7 @@ export class BalancerV2EventPool extends StatefulEventSubscriber<PoolStateMap> {
     }
 
     if (
-      subgraphPool.poolType !== BalancerPoolTypes.Weighted &&
+      !this.buySupportedPoolTypes.includes(subgraphPool.poolType) &&
```

Comment:
please reverse this check so that we first check side === SwapSide.BUY then we check if poolType supports buy or not